### PR TITLE
fix comments with dot in username

### PR DESCRIPTION
**Pull request by petrjasek**
- fix username and desk name parser to pick up these
- add a workaround for storage - it doesn't allow `.` in dict keys

SDESK-472

### CODE REVIEW

File: apps/comments/user_mentions.py [hunk 1]
```diff
@@ -23,8 +23,8 @@ def get_mentions(text):
     Returns the names of desks and users from a given text
     User names starts with '@' and desk names starts with '#'
     """
-    user_pattern = re.compile("(^|\s)\@([a-zA-Z0-9-_.]\w+)")
-    desk_pattern = re.compile("(^|\s)\#([a-zA-Z0-9-_.]\w+)")
+    user_pattern = re.compile("(^|\s)\@([a-zA-Z0-9-_.\w]+)")
+    desk_pattern = re.compile("(^|\s)\#([a-zA-Z0-9-_.\w]+)")
```
Comment (naspeh) [inline apps/comments/user_mentions.py]: `\w` consists of `a-zA-Z0-9_` so pattern can be simplified like `[\w-.]+`

File: apps/comments/comments.py [hunk 2]
```diff
@@ -42,6 +42,36 @@ class CommentsResource(Resource):
...
+    """
+    if not doc.get(field):
+        return
+    encoded = {}
+    for key, val in doc.get(field).items():
+        encoded[key.replace('.', ';')] = val
+    doc[field] = encoded
+
+
+def decode_keys(doc, field):
```
Comment (naspeh) [inline apps/comments/comments.py]: So if we'll have original symbol `;` in name it'll be replaced with `.` too, should we think about this case or not?

Comment (naspeh) [inline apps/comments/comments.py]: What do you think about to use `md5` for encoding/decoding?

Comment (petrjasek) [inline apps/comments/comments.py]: base64 sounds good, but it makes those names very cryptic, will leave it with replace for now, we don't allow `;` in username

**Comment by naspeh** [comment]
@petrjasek I've just checked, we can create desks with `;` symbols, so we need to add restrictions to desk's names or use `base64` there

**Pull request merged**